### ISSUE TITLE
setup.py?

### ISSUE THREAD

**Issue by TheChymera** [stefanSchinkel/gantt#7]
Hello, I am looking to package your software and make it available via the package manager of the Gentoo Linux (and derivative) distributions.

Commonly, for Python packages, a `setup.py` file is expected. Could you please [create one]()? If not, would you be willing to pull one if it was contributed?

**Comment by stefanSchinkel** [comment]
Dear Christian, 

thanks for considering to have my software made available via Portage. I'll try to set aside some time to write a setup.py as soon as possible. And of course, I'd me more than happy if to accept an PR for this. 
One a side note, I know next to nothing about gentoo package manager but I do have some experience in bundling python as native debian packages (using [dh-virtualenv](https://dh-virtualenv.readthedocs.io/en/1.1/#)) If that is of any use to you I could also try to get that working. 

Stefan

**Comment by TheChymera** [comment]
Ping me when you have the setup.py :)

Currently we make only https://pypi.org/project/python-gantt/ available to our users, but that package is being ill-maintained and has some issues with python3 support :-/

It does however have a lot of nice features. Personally I use gantt plots a lot, so I might be willing to contribute. Do you think the two projects could be merged? I much prefer your `.json` based approach, but really, that other package has *so many* very handy features.

**Comment by TheChymera** [comment]
@stefanSchinkel works like a charm!

Still, could you perhaps consider https://github.com/stefanSchinkel/gantt/pull/10 ? I am sed-ing this in for the Gentoo Linux package right now, but I think it is relevant for all distribution methods.

**Comment by stefanSchinkel** [comment]
@TheChymera I merged your PR but sadly it broke the CI. There seems to be an issue w/ the latest version of matplotlib. I've pinned it to the old version for the moment and everything seems to work fine. I'll still leave this issue open as I'll have to investigate on the details.

**Comment by TheChymera** [comment]
@stefanSchinkel sorry to hear that. What is the “last version” of matplotlib you're working with? I am using the following over here:

```
[I] dev-python/matplotlib
     Available versions:  2.2.2-r1^t (~)2.2.4^t (~)3.1.2^t {cairo doc examples excel gtk2 gtk3 latex qt5 test tk wxwidgets PYTHON_TARGETS="python2_7 python3_6 python3_7"}
     Installed versions:  3.1.2^t(01:25:43 AM 04/02/2020)(cairo latex wxwidgets -doc -examples -excel -gtk3 -qt5 -test -tk PYTHON_TARGETS="python3_6 -python3_7")
     Homepage:            https://matplotlib.org/
     Description:         Pure python plotting library with matlab like syntax
```

and the test suite passes nicely:

```
silenthost ~ # FEATURES=test emerge -av python-gantt

 * IMPORTANT: 6 news items need reading for repository 'gentoo'.
 * Use eselect news read to view new items.


These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R    ] dev-python/python-gantt-0.6.0_p1::chymeric  USE="test*" PYTHON_TARGETS="python3_6 -python3_7" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

Would you like to merge these packages? [Yes/No] y

>>> Verifying ebuild manifests

>>> Emerging (1 of 1) dev-python/python-gantt-0.6.0_p1::chymeric
 * 8ed033086d2974e15da61b770204e043348bb9d6.tar.gz SHA256 SHA512 WHIRLPOOL size ;-) ...                                              [ ok ]
>>> Unpacking source...
>>> Unpacking 8ed033086d2974e15da61b770204e043348bb9d6.tar.gz to /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work
>>> Source unpacked in /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work
>>> Preparing source in /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6 ...
 * Will copy sources from /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6
 * python3_6: copying to /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6-python3_6
 * python3_6: running distutils-r1_run_phase python_prepare
 * Will copy sources from /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6-python3_6/build
 * python3_6: copying to /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6-python3_6/build-python3_6
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6 ...
 * python3_6: running distutils-r1_run_phase distutils-r1_python_compile
python3.6 setup.py build -j 9
running build
>>> Source compiled.
>>> Test phase: dev-python/python-gantt-0.6.0_p1
 * python3_6: running distutils-r1_run_phase python_test
=========================================================== test session starts ============================================================
platform linux -- Python 3.6.10, pytest-5.4.1, py-1.8.0, pluggy-0.13.1 -- /usr/bin/python3.6
cachedir: .pytest_cache
rootdir: /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/work/gantt-8ed033086d2974e15da61b770204e043348bb9d6-python3_6
collecting ... collected 8 items

tests/test_basics.py::TestsPackage::testDefColor PASSED                                                                              [ 12%]
tests/test_basics.py::TestsPackage::testNegatives PASSED                                                                             [ 25%]
tests/test_basics.py::TestsPackage::testValError PASSED                                                                              [ 37%]
tests/test_basics.py::TestsPackage::testValuePassing PASSED                                                                          [ 50%]
tests/test_basics.py::TestsBasics::testPackages PASSED                                                                               [ 62%]
tests/test_basics.py::TestsBasics::testPlotTitle PASSED                                                                              [ 75%]
tests/test_basics.py::TestsBasics::testTimings PASSED                                                                                [ 87%]
tests/test_basics.py::TestsBasics::testTitle PASSED                                                                                  [100%]

============================================================ 8 passed in 0.87s =============================================================
 * python3_6: running distutils-r1_run_phase _distutils-r1_clean_egg_info
>>> Completed testing dev-python/python-gantt-0.6.0_p1

>>> Install dev-python/python-gantt-0.6.0_p1 into /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/image
 * python3_6: running distutils-r1_run_phase distutils-r1_python_install
python3.6 setup.py install --root=/var/tmp/portage/dev-python/python-gantt-0.6.0_p1/image/_python3.6
running install
running build
running install_egg_info
running egg_info
creating gantt.egg-info
writing gantt.egg-info/PKG-INFO
writing dependency_links to gantt.egg-info/dependency_links.txt
writing requirements to gantt.egg-info/requires.txt
writing top-level names to gantt.egg-info/top_level.txt
writing manifest file 'gantt.egg-info/SOURCES.txt'
reading manifest file 'gantt.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'gantt/test/test_basics.py'
warning: no files found matching 'gantt/test/basics.json'
writing manifest file 'gantt.egg-info/SOURCES.txt'
Copying gantt.egg-info to /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/image/_python3.6/usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info
running install_scripts
 * Using python3.6 in global scope
 * python3_6: running distutils-r1_run_phase distutils-r1_python_install_all
>>> Completed installing dev-python/python-gantt-0.6.0_p1 into /var/tmp/portage/dev-python/python-gantt-0.6.0_p1/image

 * Final size of build directory: 288 KiB
 * Final size of installed tree:   48 KiB

 * Verifying compiled files in /usr/lib64/python3.6/site-packages

>>> Installing (1 of 1) dev-python/python-gantt-0.6.0_p1::chymeric
 * checking 5 files for package collisions
>>> Merging dev-python/python-gantt-0.6.0_p1 to /
--- /usr/
--- /usr/lib64/
--- /usr/lib64/python3.6/
--- /usr/lib64/python3.6/site-packages/
--- /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/
>>> /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/SOURCES.txt
>>> /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/top_level.txt
>>> /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/dependency_links.txt
>>> /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/PKG-INFO
>>> /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/requires.txt
>>> Safely unmerging already-installed instance...
No package files given... Grabbing a set.
--- replaced obj /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/top_level.txt
--- replaced obj /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/requires.txt
--- replaced obj /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/dependency_links.txt
--- replaced obj /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/SOURCES.txt
--- replaced obj /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info/PKG-INFO
--- replaced dir /usr/lib64/python3.6/site-packages/gantt-1.0-py3.6.egg-info
--- replaced dir /usr/lib64/python3.6/site-packages
--- replaced dir /usr/lib64/python3.6
--- replaced dir /usr/lib64
--- replaced dir /usr
>>> Regenerating /etc/ld.so.cache...
>>> Original instance of package unmerged safely.
>>> dev-python/python-gantt-0.6.0_p1 merged.
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

 * IMPORTANT: 6 news items need reading for repository 'gentoo'.
 * Use eselect news read to view new items
```

**Comment by stefanSchinkel** [comment]
--
